controllers/components/avro_schema.go

```diff
@@ -76,8 +78,23 @@ func (as *AvroSchema) DeleteByVersion(version string) (err error) {
 	return
 }
 
-func (as *AvroSchema) CheckDeviation() (err error) {
-	return //TODO
+func (as *AvroSchema) CheckDeviation() (problem, err error) {
+	schema, err := as.SetSchemaNameNamespace()
+	if err != nil {
+		return nil, errors.Wrap(err, 0)
+	}
+
+	as.registry.Client.ResetCache()
+	srschema, err := as.registry.Client.GetLatestSchema(as.Name())
```

Comment:
Check for existence first.

Reply:
are you planning to add the existence check as part of this PR?

Reply:
yes!

Reply:
done